lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/VcStoringService.java

```diff
@@ -0,0 +1,216 @@
+                };
+
+        ipvSessionItem.addVisitedCredentialIssuerDetails(visitedCriDetails);
+    }
+
+    private VisitedCredentialIssuerDetailsDto storePendingVcs(
+            VerifiableCredentialResponse vcResponse) throws JsonProcessingException, SqsException {
+        if (!vcResponse.getUserId().equals(userId)) {
+            throw new VerifiableCredentialResponseException(
+                    HTTPResponse.SC_SERVER_ERROR,
+                    FAILED_TO_VALIDATE_VERIFIABLE_CREDENTIAL_RESPONSE);
+        }
+
+        VerifiableCredentialResponseDto vcResponseDto =
+                VerifiableCredentialResponseDto.builder()
+                        .userId(userId)
+                        .credentialStatus(VerifiableCredentialStatus.PENDING.getStatus())
+                        .build();
+
+        criResponseService.persistCriResponse(
+                userId,
+                criId,
+                OBJECT_MAPPER.writeValueAsString(vcResponseDto),
+                criOAuthSessionId,
+                CriResponseService.STATUS_PENDING);
+
+        sendAuditEventForProcessedVcResponse(
+                CriResourceRetrievedType.PENDING.getType(), criId, auditEventUser);
+
+        return new VisitedCredentialIssuerDetailsDto(criId, componentId, false, null);
+    }
+
+    private VisitedCredentialIssuerDetailsDto storeCreatedVcs(
+            VerifiableCredentialResponse vcResponse)
+            throws SqsException, ParseException, JsonProcessingException, CiPutException,
+                    CiPostMitigationsException, VerifiableCredentialResponseException {
+        var vcs = vcResponse.getVerifiableCredentials();
+
+        var issuers = new HashSet<String>();
+        for (SignedJWT vc : vcs) {
+            verifiableCredentialJwtValidator.validate(vc, criConfig, userId);
+
+            auditService.sendAuditEvent(
+                    new AuditEvent(
+                            AuditEventTypes.IPV_VC_RECEIVED,
+                            componentId,
+                            auditEventUser,
+                            getAuditExtensions(vc, VcHelper.isSuccessfulVc(vc))));
+
+            ciMitService.submitVC(vc, govukSigninJourneyId, ipAddress);
+            ciMitService.submitMitigatingVcList(
+                    List.of(vc.serialize()), govukSigninJourneyId, ipAddress);
+
+            verifiableCredentialService.persistUserCredentials(vc, criId, userId);
+
+            issuers.add(vc.getJWTClaimsSet().getIssuer());
+        }
+
+        if (vcs.isEmpty()) {
+            sendAuditEventForProcessedVcResponse(
+                    CriResourceRetrievedType.EMPTY.getType(), criId, auditEventUser);
+            issuers.add(null);
+        } else {
+            if (issuers.size() != 1) {
+                throw new VerifiableCredentialResponseException(
+                        HttpStatus.SC_BAD_GATEWAY,
+                        ErrorResponse.INVALID_NUMBER_OF_ISSUERS_IN_VC_RESPONSE);
```

Comment:
Did you add this logic?

Reply:
I did. Previously it'd take the last issuer found from an unspecified-order list, which made it seem like we expect the same issuer

Reply:
I think what we _should_ be doing is confirming that the issuer matches what we expect under the CRI ID, but let's leave that for another ticket - maybe https://govukverify.atlassian.net/browse/PYIC-3907

I don't mind if you want to leave this in, but we could probably just remove the issuer logic entirely (if it's not actually used - as below)

Reply:
I'm double checking with Chris if VisitedCredentialIssuerDetailsDto is needed anywhere which won't be removed in the next part of this ticket. If so, I'll remove all logic related to it